orcid-core/src/main/java/org/orcid/core/manager/v3/impl/ClientDetailsManagerImpl.java

```diff
@@ -421,5 +421,45 @@ public void convertPublicClientToMember(String clientId, String groupId) {
             LOGGER.info("Client {} was succesfully promoted to {}", clientId, clientType);
         }
     }
+
+
+    @Override
+    @Transactional
+    public void moveClientGroupId(String clientId, String groupId) {
+        LOGGER.info("Moving client {} as member with groupId {}", clientId, groupId);
+        ProfileEntity group = profileEntityManager.findByOrcid(groupId);
+        ClientType clientType = MemberType.PREMIUM.name().equals(group.getGroupType()) ? ClientType.PREMIUM_UPDATER : ClientType.UPDATER;
+        LOGGER.info("Client {} will be a {}", clientId, clientType);
+        // Change client type
+        if (clientDetailsDao.convertPublicClientToMember(clientId, groupId, clientType.name())) {
+            // // Change role from 'ROLE_PUBLIC' to 'ROLE_CLIENT'
+            // LOGGER.info("Updating granted authority for client {}", clientId);
+            // clientDetailsDao.updateClientGrantedAuthority(clientId, "ROLE_CLIENT");
+            // Assign scopes to client
+            List<String> clientScopes = clientScopeDao.getActiveScopes(clientId);
+            Set<String> newScopes = null;
+            if (clientType.equals(ClientType.PREMIUM_UPDATER)) {
+                newScopes = ClientType.getScopes(ClientType.PREMIUM_UPDATER);
+            } else {
+                newScopes = ClientType.getScopes(ClientType.UPDATER);
+            }
+
```

Comment:
Lets add some comments here explaining why you should delete some scopes and then add others .... Because you and me understand it but maybe not others